### PR TITLE
Faster contract group check

### PR DESCRIPTION
**Pull request by mestachs**
The new engine only used on pyramid, since the worker support both legacy and newer engine, the worker was loading 4 times the pyramid and so putting a lot of GC pressure.

I play tested this change in prod

with the modification in rails console on prod, these are the timings logged in invoicing_jobs table

duration_ms: 30924
duration_ms: 79243
duration_ms: 83865
duration_ms: 57610
duration_ms: 75326

before

duration_ms: 788701
duration_ms: 909670

an order of magnitude less.

### CODE REVIEW

File: app/controllers/setup/invoices_controller.rb [hunk 1]
```diff
@@ -56,9 +56,10 @@ def render_invoice(project, invoicing_request)
 
   def render_new_invoice(project, invoicing_request)
     options = Invoicing::InvoicingOptions.new(
-      publish_to_dhis2:       false,
-      force_project_id:       params[:simulate_draft] ? project.id : nil,
-      allow_fresh_dhis2_data: params[:simulate_draft]
+      publish_to_dhis2:             false,
+      force_project_id:             params[:simulate_draft] ? project.id : nil,
+      allow_fresh_dhis2_data:       params[:simulate_draft],
+      do_nothing_if_not_contracted: false
```
Comment (pjaspers) [inline app/controllers/setup/invoices_controller.rb]: Does that mean we're never using it? Or only not using it when rendering?

Comment (mestachs) [reply app/controllers/setup/invoices_controller.rb]: That's the problem with my chosen naming... 
This allow the invoice generation in the simulation but the worker shouldn't push dhis2 values for "non-contracted" entities

or I just reuse the publish_to_dhis2 as the same flag ?

Comment (pjaspers) [reply app/controllers/setup/invoices_controller.rb]: Might be nicer to hide it away from the user if we're already not publishing.

**Comment by pjaspers** [comment]
Number 💯 ! 🎉

**Pull request merged**